dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/master/registry/MasterRegistry.java

```diff
@@ -113,6 +110,14 @@ public void stateChanged(CuratorFramework client, ConnectionState newState) {
     public void unRegistry() {
         String address = getLocalAddress();
         String localNodePath = getMasterPath();
+        heartBeatExecutor.shutdown();
+        try {
+            if (heartBeatExecutor.awaitTermination(masterConfig.getMasterHeartbeatInterval(), TimeUnit.SECONDS)) {
+                logger.warn("The heartbeat executor does not shutdown");
+            }
+        } catch (InterruptedException e) {
+            logger.warn("The heartbeat executor is interrupted");
+        }
```

Comment:
Hi,

Good job, is it better to refer to WorkerRegistry[1] to use `heartBeatExecutor.shutdownNow()`?

[1] https://github.com/apache/incubator-dolphinscheduler/blob/dev/dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/registry/WorkerRegistry.java

Reply:
solved